azkaban-common/src/test/java/azkaban/user/FileWatcherTest.java

```diff
@@ -73,14 +72,33 @@ public void registerAndTake() throws Exception {
 
     final WatchKey key = this.fileWatcher.take();
     final List<WatchEvent<?>> events = this.fileWatcher.pollEvents(key);
-    // depending on the OS & file system there may be at least 1 or 2 events even with 1 write()
-    assertThat(events.size()).isGreaterThanOrEqualTo(1);
 
-    for (WatchEvent<?> event : events) {
+    // Since we watch the entire parent dir, *and* run multiple tests in parallel, we catch *other*
+    // files being changed here. This assert checks that at least one of them is "ours"
+    // (*and* that at least one event is fired in the first place):
+    assertTrue(events.stream().anyMatch(event -> {
```

Comment:
So... which of the four ways to say the same do we like best? In order from the closest resemblance to the old code to slightly more diverging in style. I'll leave the one with the most votes :) .

Reply:
assertTrue is maybe the worst(?) because when it fails it doesn't provide any context, it just says that true was expected but got false.

The [contribution style](https://github.com/azkaban/azkaban/blob/master/CONTRIBUTING.md#style-guides) guide says:

> Use AssertJ as the assertion library in unit tests.

Reply:
Thanks @juhoautio, I'll look for a better assert.